### ISSUE TITLE
Implement fork-choice rule

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#103]
## !! WAITING ON RELEASE CANDIDATE !!

This issue is presently on-hold until the specification is stable.

## Description

Presently, the [Beacon chain fork choice rule](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#beacon-chain-fork-choice-rule) section of the spec declares that the fork choice rule is "Latest Message Driven (LMD) Greediest Heaviest Observed SubTree (GHOST)".

I suspect there are significant changes coming to this LMD GHOST algorithm, so this issue is effectively a placeholder.

## Tasks

- Wait until the fork-choice rule is finalized.

**Comment by paulhauner** [comment]
Removing the block for the release candidate.

**Comment by paulhauner** [comment]
I am working on this

**Comment by paulhauner** [comment]
Addressed in #180. We can make a new issue to track fork choice once @AgeManning is done.
